cpp/src/fil/infer.cu

```diff
@@ -330,11 +333,14 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
     // __syncthreads() is called in infer_k
   }
   __device__ __forceinline__ void accumulate(
-    vec<NITEMS, int> single_tree_prediction, int tree) {
+    vec<NITEMS, int> single_tree_prediction, int tree, int num_rows) {
 #pragma unroll
-    for (int item = 0; item < NITEMS; ++item)
-      raft::myAtomicAdd(votes + single_tree_prediction[item] * NITEMS + item,
-                        1);
+    for (int item = 0; item < NITEMS; ++item) {
+      if (item < num_rows) {
```

Comment:
There's always enough shared memory allocated in `votes` for all `NITEMS`, independent of how many rows are actually processed. So no need for a conditional here.